### PR TITLE
Add FwupdRequestFlags for future usage

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupd/fwupd-request.c [hunk 1]
```diff
@@ -403,6 +457,100 @@ fwupd_request_set_kind(FwupdRequest *self, FwupdRequestKind kind)
...
+{
+	FwupdRequestPrivate *priv = GET_PRIVATE(self);
+	g_return_val_if_fail(FWUPD_IS_REQUEST(self), 0);
+	return priv->flags;
+}
+
+/**
+ * fwupd_request_set_flags:
+ * @self: a #FwupdRequest
+ * @flags: request flags, e.g. %FWUPD_REQUEST_FLAG_FORCE_GENERIC
```
Comment (superm1) [inline libfwupd/fwupd-request.c:481]: This flag `FWUPD_REQUEST_FLAG_FORCE_GENERIC` doesn't exist in the PR

Comment (hughsie) [reply libfwupd/fwupd-request.c]: Ta, fixed.

**Review by superm1 — APPROVED** [review]
The code looks fine, but I suggest holding off to merge it based on the comments from #5115 since I think the design changes quite a bit from those.  Maybe this is superfluous.

**Comment by hughsie** [comment]
I think we need this anyway for the daemon to signal that it's okay to substitute the plugin-generated update text with translatable text. It seems a useful thing to add regardless IMHO.

**Comment by superm1** [comment]
> I think we need this anyway for the daemon to signal that it's okay to substitute the plugin-generated update text with translatable text. It seems a useful thing to add regardless IMHO.

OK, sounds good.

**Pull request merged**